modules/api-svc/src/InboundServer/handlers.js

```diff
@@ -952,12 +959,140 @@ const putBulkTransfersByIdError = async(ctx) => {
+    });
+
+    // todo: think, what does it mean in putQuote handler!
+    //
+    //  duplicate publication until legacy code refactored
+    // await QuotesModel.triggerDeferredJob({
+    //     cache: ctx.state.cache,
+    //     message: data,
+    //     args: {
+    //         quoteId
+    //     }
+    // });
+
+    ctx.response.status = ReturnCodes.OK.CODE;
+};
+
+const postFxTransfers = async (ctx) => {
+    if (ctx.state.conf.isIsoApi) {
+        ctx.state.logger.isDebugEnabled && ctx.state.logger.push(ctx.request.body).debug('Transforming incoming ISO20022 post fxTransfers body to FSPIOP');
+        const target = await TransformFacades.FSPIOPISO20022.fxTransfers.post({ body: ctx.request.body }, { rollUpUnmappedAsExtensions: true });
+        ctx.request.body = target.body;
+    }
+
+    const { body, headers, sourceFspId } = extractBodyHeadersSourceFspId(ctx);
+    const { logger } = ctx.state;
+    const logPrefix = 'Handling POST fxTransfers request';
+
+    const model = createInboundTransfersModel(ctx);
+    model.postFxTransfers({ body, headers }, sourceFspId)
+        .then(response => logger.push({ response }).log(`${logPrefix} is done`))
+        .catch(err => logger.push({ err }).log(`${logPrefix} error`));
+
+    prepareResponse(ctx);
+};
+
+const patchFxTransfersById = async (ctx) => {
+    const req = {
+        headers: { ...ctx.request.headers },
+        data: { ...ctx.request.body }
+    };
+    const idValue = ctx.state.path.params.ID;
+
+    const model = createInboundTransfersModel(ctx);
+
+    const response = await model.sendFxPatchNotificationToBackend(req.data, idValue);
+
+    // log the result
+    ctx.state.logger.isDebugEnabled && ctx.state.logger.push({response}).
+        debug('Inbound Transfers model handled PATCH /fxTransfers/{ID} request');
+};
+
+/**
+ * Create a handler for PUT /fxTransfers/{ID} and PUT /fxTransfers/{ID}/error routes
+ *
+ * @param success {boolean} - false is for handling error callback response
+ */
+const createPutFxTransfersHandler = (success) => async (ctx) => {
+    if (ctx.state.conf.isIsoApi) {
+        const method = success ? 'put' : 'putError';
+        ctx.state.logger.isDebugEnabled && ctx.state.logger.push(ctx.request.body).debug(`Transforming incoming ISO20022 ${method} fxTransfers body to FSPIOP`);
+        const target = await TransformFacades.FSPIOPISO20022.fxTransfers[method]({ body: ctx.request.body }, { rollUpUnmappedAsExtensions: true });
+        ctx.request.body = target.body;
+    }
+
+    const { body, headers } = extractBodyHeadersSourceFspId(ctx);
+    const { ID } = ctx.state.path.params;
+
+    const channel = `${CacheKeyPrefixes.FX_TRANSFER_CALLBACK_CHANNEL}_${ID}`;
+    await ctx.state.cache.publish(channel, {
+        success,
+        data: { body, headers },
+        type: `fxTransfersResponse${success ? '' : 'Error'}`
+    });
+
+    // todo: think, what does it mean in putTransfer handler!
```

Comment:
same comment as above - not needed or does it need follow-up?

Reply:
Not needed.